tests/Application/Actions/User/UpdateUserActionTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Application\Actions\User;
+
+use App\Domain\Entity\User\Exception\UserNotFoundException;
+use App\Domain\Entity\User\User;
+use App\Domain\Entity\User\UserRepositoryInterface;
+use App\Domain\Service\User\UserEventsService;
+use DI\Container;
+use Tests\TestCase;
+
+class UpdateUserActionTest extends TestCase
+{
+    public function testActionSuccess(): void
+    {
+        $app = $this->getApp();
+        /** @var Container $container */
+        $container = $app->getContainer();
+
+        $user = new User("steve.jobs", "Steve", "Jobs");
+        $userRepository = $this->prophesize(UserRepositoryInterface::class);
+        $userEventService = $this->prophesize(UserEventsService::class);
+
+        $userRepository->findUserOfId(1)->willReturn($user);
+        $userRepository->save($user)->shouldBeCalledOnce();
+        $userEventService->userWasUpdated($user)->shouldBeCalledOnce();
+
+        $container->set(UserRepositoryInterface::class, $userRepository->reveal());
+        $container->set(UserEventsService::class, $userEventService->reveal());
+
+        $request = $this->createRequest(
+            method: "PATCH",
+            path: "/api/v1/user/1",
+            body: json_encode(["username" => "john.doe", "firstName" => "John", "lastName" => "Doe"]),
```

Comment:
In this case I would use a dataprovider for the convenience of testing different cases. https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html